### DISCUSSION TITLE
Why does jax.numpy have different trigonometric functions from numpy and math?

### DISCUSSION THREAD

**Discussion by hyunwoooh5** [jax-ml/jax#14227] [General] [answered]
While I was writing a code in jax, I found that my c++ code and python give different results. Also, I checked other libararies or and mathematica and only jax.numpy give the different values:

<img width="305" alt="image" src="https://user-images.githubusercontent.com/114940661/215848395-25c28167-f6f4-49ed-a61a-934c1d25c1c1.png">

Is there any reason for this?

**Comment by jakevdp** [comment, marked as answer]
JAX defaults to 32-bit computation (see http://go/jax-sharp-bits#double-64bit-precision), while `math`, `numpy`, and others default to 64-bit computation. If you enable 64-bit computation in JAX, then you should see that the results match:
```python
import jax
jax.config.update('jax_enable_x64', True)

import math
import numpy as np
import jax.numpy as jnp

print(f"{jnp.cos(3.0):.15f}")
print(f"{math.cos(3.0):.15f}")
print(f"{np.cos(3.0):.15f}")
```
```
-0.989992496600445
-0.989992496600445
-0.989992496600445
```
